### PR TITLE
Create CMS Mapping for Sections

### PR DESCRIPTION
**Pull request by gvc**
## What's the purpose of this pull request?

Add CMS mapping for all default sections in our starter.

## How to test it?

Install this version of Faststore on your [starter](https://github.com/vtex-sites/starter.store), run `vtex cms sync` and see if everything is in place or use the preview link.

[Link Preview](https://arthurspace2--storeframework.myvtex.com/admin/new-cms/faststore)

### CODE REVIEW

File: packages/core/cms/sections.json [hunk 1]
```diff
@@ -1,4 +1,267 @@
...
+            "clear-button-label": {
+              "title": "Clear Button Label",
+              "type": "string",
+              "default": "Clear"
+            }
+          }
+        },
+        "search-history": {
+          "type": "object",
+          "title": "History",
```
Comment (hellofanny) [inline packages/core/cms/sections.json:49]:
```suggestion
        "title": "History",
        "type": "object",
```

File: packages/core/cms/sections.json [hunk 2]
```diff
@@ -1,4 +1,267 @@
...
+              "type": "string",
+              "default": "Clear History"
+            },
+            "icon": {
+              "type": "object",
+              "properties": {
+                "icon": {
+                  "title": "Icon",
+                  "type": "string",
+                  "enumNames": ["REPLACe"],
```
Comment (hellofanny) [inline packages/core/cms/sections.json:67]: Replace.

Comment (gvc) [reply packages/core/cms/sections.json]: I have replaced 😆

File: packages/core/cms/sections.json [hunk 3]
```diff
@@ -46,30 +326,253 @@
...
+                ]
+              },
+              "alt": {
+                "title": "Alternative Label",
+                "type": "string"
+              }
+            }
+          }
+        },
+        "footer-links": {
```
Comment (hellofanny) [inline packages/core/cms/sections.json:384]: Should this be required?

File: packages/core/cms/sections.json [hunk 4]
```diff
@@ -46,30 +326,253 @@
...
+              }
+            },
+            "alt": {
+              "type": "string",
+              "title": "Alternative Label"
+            }
+          }
+        },
+        "copyright-info": {
+          "title": "Copyright message",
```
Comment (hellofanny) [inline packages/core/cms/sections.json:493]:
```suggestion
          "title": "Copyright Message",
```

File: packages/core/cms/sections.json [hunk 5]
```diff
@@ -1,4 +1,283 @@
...
+          "properties": {
+            "regionalization": {
+              "type": "object",
+              "title": "Regionalization",
+              "properties": {
+                "enabled": {
+                  "type": "boolean",
+                  "title": "Use Regionalization?",
+                  "default": true
+                },
```
Comment (eduardoformiga) [inline packages/core/cms/sections.json:230]: We don't have this option in frontend, should we create a task to handle it?

File: packages/core/cms/sections.json [hunk 6]
```diff
@@ -46,30 +325,253 @@
       }
     }
   },
+  {
+    "name": "Footer",
+    "schema": {
+      "title": "Footer",
+      "description": "Footer displayed on all pages",
+      "type": "object",
+      "required": [],
```
Comment (eduardoformiga) [inline packages/core/cms/sections.json:334]: we don't pass required props. Can we remove this line or is it required?

File: packages/core/cms/sections.json [hunk 7]
```diff
@@ -46,30 +325,253 @@
...
+                "required": ["icon", "alt"],
+                "properties": {
+                  "icon": {
+                    "type": "object",
+                    "properties": {
+                      "icon": {
+                        "title": "Icon",
+                        "type": "string",
+                        "enumNames": ["Replace"],
+                        "enum": ["REPLACE"]
```
Comment (eduardoformiga) [inline packages/core/cms/sections.json:525]: should we add a real example here instead of `Replace`?

File: packages/core/cms/sections.json [hunk 8]
```diff
@@ -233,18 +767,39 @@
             "properties": {
               "key": {
                 "title": "Key",
-                "description": "Tip: For collections, use: productClusterIds",
+                "hint": "For collections, use: productClusterIds",
```
Comment (eduardoformiga) [inline packages/core/cms/sections.json:770]: out of curiosity: where I can see all the available key types (like this `hint` option? I also didn't find this hint in the CMS (Admin).
Screenshot 2023-04-04 at 18 33 09

File: packages/core/cms/sections.json [hunk 9]
```diff
@@ -233,18 +767,39 @@
...
                 "type": "string",
                 "default": "productClusterIds"
               },
               "value": {
                 "title": "Value",
-                "description": "Tip: For collection 140, use: 140",
+                "hint": "For collection 140, use: 140",
                 "type": "string",
                 "default": "140"
```
Comment (eduardoformiga) [inline packages/core/cms/sections.json:778]: Where can wee see this magic number mapping? how can we know that `140` means collection? Would be good a way to see all available clusters.

File: packages/core/cms/sections.json [hunk 10]
```diff
@@ -233,18 +767,39 @@
...
+            "show-discount-badge": {
+              "title": "Show discount badge?",
+              "type": "boolean",
+              "default": true
+            },
+            "show-buy-button": {
+              "title": "Show buy button?",
+              "type": "boolean",
+              "default": true
+            },
```
Comment (eduardoformiga) [inline packages/core/cms/sections.json:796]: I think we don't have this checking in frontend. 👀

**Review by eduardoformiga — APPROVED** [review]
Good job! 👏👏👏 
I left some comments along the review. 

Some CMS improvements that I got along the way and we can discuss with CMS team:
- Create a way to Search Sections in CMS admin instead of scroll and find.
- Validate input types (integer, URL etc).

File: packages/core/cms/sections.json (hunk 3 above)
Comment (ArthurTriis1) [reply packages/core/cms/sections.json:383]: Not necessarily, without links, the layout doesn't broke

File: packages/core/cms/sections.json (hunk 5 above)
Comment (ArthurTriis1) [reply packages/core/cms/sections.json:230]: Oh, it is true, I created in this link as subtask of navigation https://vtex-dev.atlassian.net/browse/FS-983

File: packages/core/cms/sections.json (hunk 8 above)
Comment (ArthurTriis1) [reply packages/core/cms/sections.json:770]: I changed to description, thanks!

Screenshot 2023-04-05 at 12 17 46

File: packages/core/cms/sections.json (hunk 10 above)
Comment (ArthurTriis1) [reply packages/core/cms/sections.json:796]: We control this behavior using the OnClick Button method, who to integrate this component can improve

File: packages/core/cms/sections.json (hunk 9 above)
Comment (ArthurTriis1) [reply packages/core/cms/sections.json:778]: We can see in Collections page, I will improve the description

**Pull request merged**